src/test_custom_request_schedule_manager.cc

```diff
@@ -37,82 +38,81 @@ namespace triton::perfanalyzer {
 class TestCustomRequestScheduleManager : public TestLoadManagerBase,
                                          public CustomRequestScheduleManager {
  public:
-  TestCustomRequestScheduleManager(
-      PerfAnalyzerParameters params, bool is_sequence_model = false,
-      bool is_decoupled_model = false, bool use_mock_infer = false)
-      : use_mock_infer_(use_mock_infer),
-        TestLoadManagerBase(params, is_sequence_model, is_decoupled_model),
-        CustomRequestScheduleManager(params, GetParser(), GetFactory()),
-        schedule_(params.schedule)
+  TestCustomRequestScheduleManager(PerfAnalyzerParameters params)
+      : TestLoadManagerBase(params, false, false),
+        CustomRequestScheduleManager(params, GetParser(), GetFactory())
   {
   }
 
-  void TestSchedule(PerfAnalyzerParameters params)
+  void TestSchedule(const std::vector<std::chrono::milliseconds> schedule)
   {
+    schedule_ = schedule;
     int request_count = schedule_.size();
     PauseWorkers();
     ConfigureThreads(request_count);
     GenerateSchedule();
 
-    std::vector<std::chrono::nanoseconds> expected_timestamps;
-    std::chrono::nanoseconds timestamp(0);
-
-    for (float schedule_value : schedule_) {
-      timestamp = std::chrono::duration_cast<std::chrono::nanoseconds>(
-          std::chrono::duration<float>(schedule_value));
+    std::vector<std::chrono::nanoseconds> expected_timestamps{};
 
+    for (const auto& timestamp_ms : schedule_) {
+      const auto timestamp{
+          std::chrono::duration_cast<std::chrono::nanoseconds>(timestamp_ms)};
       expected_timestamps.push_back(timestamp);
     }
 
-    size_t request_index = 0;
-    for (auto worker : workers_) {
-      auto timestamp = std::dynamic_pointer_cast<RequestRateWorker>(worker)
-                           ->GetNextTimestamp();
-      REQUIRE(timestamp.count() == expected_timestamps[request_index].count());
-      request_index++;
+    std::vector<std::chrono::nanoseconds> actual_timestamps{};
+
+    for (const auto& iworker : workers_) {
+      const auto worker{std::dynamic_pointer_cast<RequestRateWorker>(iworker)};
+      for (size_t i{0}; i < worker->schedule_->intervals.size(); ++i) {
+        const auto timestamp{worker->GetNextTimestamp()};
+        actual_timestamps.push_back(timestamp);
+      }
     }
-  }
```

Comment:
It seems like there was a bug here (or I just don't understand it).

Old unit test was looping through each worker and grabbing the first timestamp from each other? Seems like that wouldn't check the entire schedule if any worker had more than one piece of the schedule.

The new unit test loops through each worker and grabs all pieces of the schedule that it owns and groups it all and sorts it and compares with expected schedule.